src/main/java/org/jabref/gui/util/BaseDialog.java

```diff
@@ -17,6 +18,13 @@
 public class BaseDialog<T> extends Dialog<T> implements org.jabref.gui.Dialog<T> {
 
     protected BaseDialog() {
+        getDialogPane().getScene().addEventFilter(KeyEvent.KEY_RELEASED, event -> {
+            KeyBindingRepository keyBindingRepository = Globals.getKeyPrefs();
+            if (keyBindingRepository.checkKeyCombinationEquality(KeyBinding.CLOSE, event)) {
+                close();
+            }
+        });
+
         getDialogPane().getScene().setOnKeyPressed(event -> {
             KeyBindingRepository keyBindingRepository = Globals.getKeyPrefs();
             if (keyBindingRepository.checkKeyCombinationEquality(KeyBinding.CLOSE, event)) {
```

Comment:
You are probably going to have to delete this if statement

Reply:
Yeah, that makes perfect sense, @k3KAW8Pnf7mkmdSMPHz27 - I can see how it might cause undesirable behaviour. I will take a look and see if I can find a 'non-terrible way' (XD) to resolve it.